format/macho/macho.go

```diff
@@ -0,0 +1,574 @@
+	LC_SYMTAB:                   "symtab",
+	LC_SYMSEG:                   "symseg",
+	LC_THREAD:                   "thread",
+	LC_UNIXTHREAD:               "unixthread",
+	LC_LOADFVMLIB:               "loadfvmlib",
+	LC_IDFVMLIB:                 "idfvmlib",
+	LC_IDENT:                    "ident",
+	LC_FVMFILE:                  "fvmfile",
+	LC_PREPAGE:                  "prepage",
+	LC_DYSYMTAB:                 "dysymtab",
+	LC_LOAD_DYLIB:               "load_dylib",
+	LC_ID_DYLIB:                 "id_dylib",
+	LC_LOAD_DYLINKER:            "load_dylinker",
+	LC_ID_DYLINKER:              "id_dylinker",
+	LC_PREBOUND_DYLIB:           "prebound_dylib",
+	LC_ROUTINES:                 "routines",
+	LC_SUB_FRAMEWORK:            "sub_framework",
+	LC_SUB_UMBRELLA:             "sub_umbrella",
+	LC_SUB_CLIENT:               "sub_client",
+	LC_SUB_LIBRARY:              "sub_library",
+	LC_TWOLEVEL_HINTS:           "twolevel_hints",
+	LC_PREBIND_CKSUM:            "prebind_cksum",
+	LC_LOAD_WEAK_DYLIB:          "load_weak_dylib",
+	LC_SEGMENT_64:               "segment_64",
+	LC_ROUTINES_64:              "routines_64",
+	LC_UUID:                     "uuid",
+	LC_RPATH:                    "rpath",
+	LC_CODE_SIGNATURE:           "code_signature",
+	LC_SEGMENT_SPLIT_INFO:       "segment_split_info",
+	LC_REEXPORT_DYLIB:           "reexport_dylib",
+	LC_LAZY_LOAD_DYLIB:          "lazy_load_dylib",
+	LC_ENCRYPTION_INFO:          "encryption_info",
+	LC_DYLD_INFO:                "dyld_info",
+	LC_DYLD_INFO_ONLY:           "dyld_info_only",
+	LC_LOAD_UPWARD_DYLIB:        "load_upward_dylib",
+	LC_VERSION_MIN_MACOSX:       "version_min_macosx",
+	LC_VERSION_MIN_IPHONEOS:     "version_min_iphoneos",
+	LC_FUNCTION_STARTS:          "function_starts",
+	LC_DYLD_ENVIRONMENT:         "dyld_environment",
+	LC_MAIN:                     "main",
+	LC_DATA_IN_CODE:             "data_in_code",
+	LC_SOURCE_VERSION:           "source_version",
+	LC_DYLIB_CODE_SIGN_DRS:      "dylib_code_sign_drs",
+	LC_ENCRYPTION_INFO_64:       "encryption_info_64",
+	LC_LINKER_OPTION:            "linker_option",
+	LC_LINKER_OPTIMIZATION_HINT: "linker_optimization_hint",
+	LC_VERSION_MIN_TVOS:         "version_min_tvos",
+	LC_VERSION_MIN_WATCHOS:      "version_min_watchos",
+	LC_NOTE:                     "note",
+	LC_BUILD_VERSION:            "build_version",
+}
+
+func machoDecode(d *decode.D, in interface{}) interface{} {
+	var archBits int
+	var ncmds uint64
+	magicBuffer := d.U32LE()
+
+	if magicBuffer == MH_MAGIC || magicBuffer == MH_MAGIC_64 {
+		d.Endian = decode.LittleEndian
+		if magicBuffer == MH_MAGIC {
+			archBits = 32
+		} else {
+			archBits = 64
+		}
+	} else if magicBuffer == MH_CIGAM || magicBuffer == MH_CIGAM_64 {
+		d.Endian = decode.BigEndian
+		if magicBuffer == MH_CIGAM {
+			archBits = 32
+		} else {
+			archBits = 64
+		}
+	} else {
+		d.Fatalf("Invalid magic field")
+	}
+
+	d.SeekAbs(0)
+	d.FieldStruct(fmt.Sprintf("mach_header_%d", archBits), func(d *decode.D) {
```

Comment:
Skip "mach_"? btw what do you think about just calling it "header" and then add archBits as a value field?

Reply:
Done.